action.yml

```
@@ -23,12 +23,15 @@ branding:
 runs:
   using: "composite"
   steps:
-    - name: Set remote URL
+    - name: Attribute commits to last committer
       run: |
-        # Attribute commits to the last committer on HEAD
         git config --global user.email "$(git log -1 --pretty=format:'%ae')"
         git config --global user.name "$(git log -1 --pretty=format:'%an')"
-        git remote set-url origin "https://x-access-token:${{ github.token }}@github.com/$GITHUB_REPOSITORY"
+      shell: bash
+    - name: Configure temporary Git credentials
+      run: |
+        git config --local credential.helper 'cache --timeout=300'
+        echo -e "protocol=https\nhost=github.com\nusername=x-access-token\npassword=${{ github.token }}" | git credential approve
```

I've tested this [in another repository](https://github.com/landons-test-org/testing-gits-credential-helper/blob/a53db38664a05390dd2524d6497712d284ce0524/.github/workflows/test-credential-helper.yml#L16-L33). Here's [the output](https://github.com/landons-test-org/testing-gits-credential-helper/actions/runs/16145060646/job/45561582262#step:4:1)

<details><summary>Logs from Actions output in case review is completed after logs age out</summary>
<p>

**Note:** The workflow file in this instance is `test-credential-helper`.

```plaintext
=== Checking .git/config ===
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://github.com/landons-test-org/testing-gits-credential-helper
	fetch = +refs/heads/*:refs/remotes/origin/*
[gc]
	auto = 0
[branch "main"]
	remote = origin
	merge = refs/heads/main
[credential]
	helper = cache --timeout=300
\n=== Checking ~/.gitconfig ===
cat: /home/runner/.gitconfig: No such file or directory
No ~/.gitconfig found

=== Checking git remote URLs ===
origin	https://github.com/landons-test-org/testing-gits-credential-helper (fetch)
origin	https://github.com/landons-test-org/testing-gits-credential-helper (push)

=== Checking for credential files ===
./.github/workflows/test-credential-helper.yml
```

</p>
</details>

---

But is the credential cache running already? Per my local testing, it is not started by those two commands and it silently drops the credential approval.

---

This was a good catch 😅

I've done some more reading and _think_ I've got [an approach](https://github.com/landons-test-org/testing-gits-credential-helper/blob/1ac63486116b6dbc982be5949d3a116cd0325cab/.github/workflows/test-credential-helper.yml#L21-L28) that does what we need. Here's [evidence of the daemon running](https://github.com/landons-test-org/testing-gits-credential-helper/actions/runs/16170048629/job/45641039599#step:5:5).

---

I've run with ☝️ approach in [b12ac8b](https://github.com/rubygems/release-gem/pull/20/commits/b12ac8bdcb07a6fa97f39cf5cc87494a8f26aac5)